mineinabyss-features/src/main/kotlin/com/mineinabyss/armorstandlock/ArmorStandLockingFeature.kt

```diff
@@ -0,0 +1,118 @@
+                            val locked = entity?.toGeary()?.get<LockArmorStand>() ?: return@playerAction
+
+                            locked.lockState = !locked.lockState
+                            if (locked.lockState) player.success("This armor stand is now protected!")
+                            else player.error("This armor stand is no longer protected!")
+                        }
+                    }
+                    "add"(desc = "Add a player to this armor stand.") {
+                        val playerName by stringArg()
+                        playerAction {
+                            val entity = player.playerData.recentRightclickedEntity
+                            val locked = entity?.toGeary()?.get<LockArmorStand>() ?: return@playerAction
+                            val uuid = Bukkit.getOfflinePlayer(playerName).uniqueId
+
+                            if (locked.allowedAccess.contains(uuid)) {
+                                player.error("This player can already interact with this armor stand")
+                                return@playerAction
+                            }
+                            else {
+                                locked.allowedAccess += uuid
+                                player.success("$playerName can now interact with this armor stand")
+                            }
+                        }
+                    }
+
+                    "remove"(desc = "Remove a player to this armor stand.") {
+                        val playerName by stringArg {
+                            parseErrorMessage = { "No player with name: $passed." }
+                        }
+                        playerAction {
+                            val entity = player.playerData.recentRightclickedEntity
+                            val locked = entity?.toGeary()?.get<LockArmorStand>() ?: return@playerAction
+                            val uuid = Bukkit.getOfflinePlayer(playerName).uniqueId
+
+                            if (locked.allowedAccess.contains(uuid)) {
+                                locked.allowedAccess -= uuid
+                                player.success("$playerName has been removed from this armor stand")
+                                return@playerAction
+                            }
+                            player.error("This player cannot interact with this armor stand")
+                        }
+                    }
+
+                    "clear"(desc = "Clear all other players from this armor stand.") {
+                        playerAction {
+                            val entity = player.playerData.recentRightclickedEntity
+                            val locked = entity?.toGeary()?.get<LockArmorStand>() ?: return@playerAction
+                            val owner = locked.owner
+
+                            locked.allowedAccess.forEach {
+                                if (it != owner) locked.allowedAccess.minus(it)
```

Comment:
Also making a new list, make it mutable and you should just call `clear` then add the owner back